cmd/notation/blob/inspect.go

```diff
@@ -0,0 +1,157 @@
+	MediaType     string
+}
+
+func inspectCommand(opts *blobInspectOpts) *cobra.Command {
+	if opts == nil {
+		opts = &blobInspectOpts{}
+	}
+	longMessage := `Inspect all signatures associated with the signed artifact.
+
+Example - Inspect signatures on an BLOB artifact:
+  notation blob inspect <signature_path>
+
+Example - Inspect signatures on an BLOB artifact output as json:
+  notation blob inspect --output json <signature_path>
+`
+
+	command := &cobra.Command{
+		Use:   "blob inspect [signaturePath]",
+		Short: "Inspect all signatures associated with the signed artifact",
+		Long:  longMessage,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return errors.New("missing signature path to the artifact: use `notation blob inspect --help` to see what parameters are required")
+			}
+			opts.signaturePath = args[0]
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runBlobInspect(cmd, opts)
+		},
+	}
+
+	opts.LoggingFlagOpts.ApplyFlags(command.Flags())
+	cmd.SetPflagOutput(command.Flags(), &opts.outputFormat, cmd.PflagOutputUsage)
+	return command
+}
+
+func runBlobInspect(command *cobra.Command, opts *blobInspectOpts) error {
+	// set log level
+	ctx := opts.LoggingFlagOpts.InitializeLogger(command.Context())
+
+	if opts.outputFormat != cmd.OutputJSON && opts.outputFormat != cmd.OutputPlaintext {
+		return fmt.Errorf("unrecognized output format %s", opts.outputFormat)
+	}
+
+	output := sharedutils.InspectOutput{MediaType: opts.MediaType, Signatures: []sharedutils.SignatureOutput{}}
+	//Added sigRepo and desc as placeholders. Once notation-go changes are merged revisit and update FetchSignatureBlob
+	sigBlob, _, _ := opts.sigRepo.FetchSignatureBlob(ctx, opts.desc)
+	sigEnvelope, _ := signature.ParseEnvelope(opts.MediaType, sigBlob)
+	envelopeContent, _ := sigEnvelope.Content()
+	signedArtifactDesc, _ := envelope.DescriptorFromSignaturePayload(&envelopeContent.Payload)
+	signatureAlgorithm, _ := proto.EncodeSigningAlgorithm(envelopeContent.SignerInfo.SignatureAlgorithm)
+
+	sig := sharedutils.SignatureOutput{
+		SignatureAlgorithm:    string(signatureAlgorithm),
+		SignedAttributes:      sharedutils.GetSignedAttributes(opts.outputFormat, envelopeContent),
+		UserDefinedAttributes: signedArtifactDesc.Annotations,
+		UnsignedAttributes:    sharedutils.GetUnsignedAttributes(envelopeContent),
+		Certificates:          sharedutils.GetCertificates(opts.outputFormat, envelopeContent),
+		SignedArtifact:        *signedArtifactDesc,
+	}
+
+	// clearing annotations from the SignedArtifact field since they're already
+	// displayed as UserDefinedAttributes
+	sig.SignedArtifact.Annotations = nil
+
+	output.Signatures = append(output.Signatures, sig)
+	blobPath := "placeholder"
+	if err := printOutput(opts.outputFormat, blobPath, output); err != nil {
+		return err
+	}
+	return nil
+}
+
+func printOutput(outputFormat string, blobPath string, output sharedutils.InspectOutput) error {
```

Comment:
Code is duplicated in other `inspect.go`

Reply:
Made changes and fixed them